### PR TITLE
Update k8s.io deps.

### PR DESCRIPTION
**Pull request by hpidcock**
## Update k8s.io deps

- Update k8s.io deps (plus others that conflict with updated k8s deps)
- Move some constants and utility functions into
caas/provider/{constants,watcher,utils}

## QA steps

Unit tests

## Documentation changes

N/A

## Bug reference

N/A

### CODE REVIEW

File: caas/kubernetes/provider/constants/constants.go [hunk 1]
```diff
@@ -3,6 +3,12 @@
 
 package constants
 
+import (
+	"github.com/juju/juju/agent"
```
Comment (ycliuhw) [inline caas/kubernetes/provider/constants/constants.go:7]: imports order

File: caas/kubernetes/provider/operator_upgrade.go [hunk 2]
```diff
@@ -15,11 +16,13 @@ import (
 	core "k8s.io/api/core/v1"
 	k8serrors "k8s.io/apimachinery/pkg/api/errors"
 	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
```
Comment (ycliuhw) [inline caas/kubernetes/provider/operator_upgrade.go:19]: meta "k8s.io/apimachinery/pkg/apis/meta/v1"
v1 "k8s.io/apimachinery/pkg/apis/meta/v1"

are they the same?

File: caas/kubernetes/provider/operator_upgrade_test.go [hunk 3]
```diff
@@ -15,6 +16,7 @@ import (
 	apps "k8s.io/api/apps/v1"
 	core "k8s.io/api/core/v1"
 	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
```
Comment (ycliuhw) [inline caas/kubernetes/provider/operator_upgrade_test.go:19]: same here

File: caas/kubernetes/provider/upgrade.go [hunk 4]
```diff
@@ -4,14 +4,18 @@
...
 import (
+	"context"
+
 	"github.com/juju/errors"
 	"github.com/juju/names/v4"
 	"github.com/juju/version"
 	core "k8s.io/api/core/v1"
 	k8serrors "k8s.io/apimachinery/pkg/api/errors"
 	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
```
Comment (ycliuhw) [inline caas/kubernetes/provider/upgrade.go:15]: same here

File: caas/kubernetes/provider/watcher/test/helpers.go [hunk 5]
```diff
@@ -1,17 +1,17 @@
 // Copyright 2020 Canonical Ltd.
 // Licensed under the AGPLv3, see LICENCE file for details.
 
-package provider_test
+package k8swatchertest
 
 import (
 	jujuclock "github.com/juju/clock"
-	"github.com/juju/juju/caas/kubernetes/provider"
+	k8swatcher "github.com/juju/juju/caas/kubernetes/provider/watcher"
```
Comment (ycliuhw) [inline caas/kubernetes/provider/watcher/test/helpers.go:8]: import sort

**Comment by hpidcock** [comment]
$$merge$$

**Pull request merged**